File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/RolePlayer/RolePlayerBankingDetail.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.ClientCare.Contracts.Entities.RolePlayer
{
    public class RolePlayerBankingDetail
    {
        public int RolePlayerId { get; set; } // RolePlayerId
        public int PurposeId { get; set; } // PurposeId
        public System.DateTime EffectiveDate { get; set; } // EffectiveDate
        public string AccountNumber { get; set; } // AccountNumber (length: 50)
        public int BankBranchId { get; set; } // BankBranchId
        public BankAccountTypeEnum BankAccountType { get; set; } // BankAccountTypeId
        public string AccountHolderName { get; set; } // AccountHolderName (length: 255)
        public string AccountHolderIdNumber { get; set; } // AccountHolderIdNumber (length: 50)
        public string BranchCode { get; set; } // BranchCode (length: 50)
        public string ApprovalRequestedFor { get; set; } // ApprovalRequestedFor (length: 50)
        public int? ApprovalRequestId { get; set; } // ApprovalRequestId
        public bool? IsApproved { get; set; } // IsApproved
        public string Reason { get; set; } // Reason (length: 255)
        public bool IsDeleted { get; set; } // IsDeleted
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public System.DateTime CreatedDate { get; set; } // CreatedDate
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public System.DateTime ModifiedDate { get; set; } // ModifiedDate
        public int RolePlayerBankingId { get; set; } // RolePlayerBankingId (Primary key)
        public string BankName { get; set; }
        public bool IsForeign { get; set; }
        public string BankBranchName { get; set; }
        public string AccountType { get; set; }
        public string Initials { get; set; }
    }
}

File: Web/RMA-Member-Portal/memberportal-app/src/app/shared/models/role-player-banking-detail.ts
import { BankAccountTypeEnum } from "../enums/bank-account-type-enum";

export class RolePlayerBankingDetail {
  rolePlayerId: number;
  purposeId: number;
  effectiveDate: Date;
  accountNumber: string;
  bankBranchId: number;
  bankAccountType: BankAccountTypeEnum;
  accountHolderName: string;
  initials: string;
  accountHolderIdNumber: string;
  branchCode: string;
  approvalRequestedFor: string;
  approvalRequestId: number;
  isApproved?: boolean;
  reason: string;
  isDeleted: boolean;
  createdBy: string;
  createdDate: Date;
  modifiedBy: string;
  modifiedDate: Date;
  
  rolePlayerBankingId: number;
  bankName: string;

  accountHolderSurname: string;
  bankId: number;
  account: string;
  statusText: string;
  accountHolder: string;
  idNumber: string;
}
